src/EconomicStrategy/EconomicStrategyHelpers.ts

```diff
@@ -71,4 +80,64 @@ const shouldClaimTx = async (txRequest: any, config: Config) => {
   return profitable && enoughBalance && !exceedsDepositLimit;
 };
 
-export { shouldClaimTx };
+/**
+ * Calculates the correct gas price to use for execution, taking into consideration
+ * the economicStrategy `maxGasSubsidy` and the current network conditions.
+ * @param {TransactionRequest} txRequest Transaction Request object to check.
+ * @param {Config} config Configuration object.
+ */
+const getExecutionGasPrice = async (txRequest: any, config: Config): Promise<BigNumber> => {
+  const currentNetworkPrice = await config.util.networkGasPrice();
+
+  if (!config.economicStrategy) {
+    return currentNetworkPrice;
+  }
+
+  const maxGasSubsidy = config.economicStrategy.maxGasSubsidy;
+
+  if (typeof maxGasSubsidy !== 'undefined' && maxGasSubsidy !== null) {
+    const minGasPrice = txRequest.gasPrice;
+    const maxGasPrice = minGasPrice.plus(minGasPrice.times(maxGasSubsidy / 100));
+
+    if (currentNetworkPrice.lessThan(minGasPrice)) {
+      return minGasPrice;
+    } else if (
+      currentNetworkPrice.greaterThanOrEqualTo(minGasPrice) &&
+      currentNetworkPrice.lessThan(maxGasPrice)
+    ) {
+      return currentNetworkPrice;
+    } else if (currentNetworkPrice.greaterThanOrEqualTo(maxGasPrice)) {
+      return maxGasPrice;
+    }
+  }
+
+  return currentNetworkPrice;
+};
+
+/**
+ * Checks if the transaction is profitable to be executed when considering the
+ * current network gas prices.
+ * @param {TransactionRequest} txRequest Transaction Request object to check.
+ * @param {Config} config Configuration object.
+ */
+const shouldExecuteTx = async (txRequest: any, config: Config): Promise<boolean> => {
+  const isClaimedByMe = config.wallet.getAddresses().indexOf(txRequest.claimedBy) !== -1;
+
+  const gasPrice = await this.getExecutionGasPrice(txRequest, config);
+  const gasAmount = config.util.calculateGasAmount(txRequest);
+  const reimbursement = txRequest.gasPrice.times(gasAmount);
+  const deposit = isClaimedByMe ? txRequest.requiredDeposit : new BigNumber(0);
+
+  const paymentModifier = await txRequest.claimPaymentModifier();
+  const reward = txRequest.bounty.times(paymentModifier);
+
+  const gasCost = gasPrice.times(gasAmount);
+
+  if (gasCost.greaterThan(deposit.plus(reward).plus(reimbursement))) {
```

Comment:
Please move this `deposit.plus(reward).plus(reimbursement)` to separate variable. Also add `debug` logging with gasCost and 'expectedReward' (or whatever name for above)

Reply:
Added this in `Added logging to shouldExecuteTx` commit